unity/unity-embed-host/UnsafeExtensions.cs

```diff
@@ -16,43 +16,28 @@ namespace Unity.CoreCLRHelpers;
 /// </summary>
 static class UnsafeExtensions
 {
-#if TESTING_UNITY_CORECLR
-    private static readonly ConcurrentBag<GCHandle> handles = new ();
-#endif
-
     public static object ToManagedRepresentation(this nint intPtr)
-        => Unsafe.As<nint, object>(ref intPtr);
-
-    public static nint ToNativeRepresentation(this object obj)
     {
-#if TESTING_UNITY_CORECLR
-        WorkaroundToGetGCSafety(obj);
-#endif
-        return Unsafe.As<object, nint>(ref obj);
+        var value = intPtr;
+        if (IsHandle(value))
+        {
+            var handle = (nuint)value & ~kBitMask;
+            return ((GCHandle)(nint)handle).Target;
+        }
+        // legacy pointer code path until we are fully GC safe
+        return Unsafe.As<nint, object>(ref intPtr);
     }
 
-#if TESTING_UNITY_CORECLR
-    static void WorkaroundToGetGCSafety(object obj)
+    public static nint ToNativeRepresentation(this object obj)
     {
-        var valueToPin = 1;
-        var handle = GCHandle.Alloc(valueToPin, GCHandleType.Pinned);
```

Comment:
I see these were previously pinned, but I don't recall why that was needed @mrvoorhe ?

Reply:
Because we cached these in a static concurrent bag to ensure GC safety during tests.

Reply:
I think we are fine to remove this. With the UnityGC I this should not be needed.  When CoreCLR GC is enabled, we need to fix the tests anyway.  + The tests may be replaced with managed soon.

Reply:
I missed your comment @bholmes .  This is for the managed embedding tests that call into native.  I wasn't able to get the null gc working for linux and mac.  https://github.com/Unity-Technologies/runtime/blob/81ca339de14bd78c1f527946f5161c604554c491/unity/CITools/BuildDriver/EmbeddingHost.cs#L201